source/extensions/tracers/xray/tracer.cc

```diff
@@ -94,6 +95,11 @@ void Span::finishSpan() {
     s.mutable_annotations()->insert({item.first, item.second});
   }
 
+  // Gets this Span's direction from operation name, direction is either
+  // "ingress" or "egress" and will be the first word from operation_name_
+  s.mutable_annotations()->insert(
+      {DirectionKey, operation_name_.substr(0, operation_name_.find(" "))});
```

Comment:
Are we certain that there will be a space in the `operation_name_`.  I hate to suggest using a regex to extract the direction but that feels more exact.  If not, I think we need a bit more logic around the string operations here in case we end up with data that we don't expect.

Also, can we add a comment with the data we expect to see in `operation_name_` here?

Reply:
This `operation_name_` value is coming from common code and is nothing but `span_name` that is populated there. https://github.com/envoyproxy/envoy/blob/main/source/common/tracing/http_tracer_impl.cc#L266-L273

`operation_name_` can have only two possible values. 
```
operation_name_ = "ingress";
operation_name_ = "egress color.howto-ecs-basics.mesh.local:8080"; // egress <header_hostname>
```
So what this substr function will do is just look for the first word which will always be either `ingress` or `egress`. Even if the space char is not present in this `operation_name_` variable we will still get the first word.

So, I hope we can still avoid the regex for this simple usecase? I just update the code comment for now.

Reply:
Made this change. Not checking for `std::string::npos` as suggested.